cluster/kubernetes/resource/resource.go

```diff
@@ -36,6 +37,19 @@ func (o *baseObject) debyte() {
 	o.bytes = nil
 }
 
+// TODO: Implement this for more service types
+func (o baseObject) ServiceIDs() []flux.ServiceID {
```

Comment:
> though we have gone to lengths elsewhere to avoid that assumption.

sounds sensible

> Can we co-opt FindDefinedServices here to get the list of services affected?

yes and no. figuring out which service a deployment belongs to (same for configmaps, etc), means we need access to all the files, so we can scan the service's selectors for ones which match. Wiring it through is quite a large change.